### PR TITLE
incept container API definition

### PR DESCRIPTION
**Pull request by TomasTomecek**
not complete yet, comments are welcome

### CODE REVIEW

File: conu/apidefs/container.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    def name(self):
+        """
+        Return name of this container.
+
+        :return: str
+        """
+        raise NotImplementedError("name method is not implemented")
+
+    def get_ip(self):
```
Comment (bkabrda) [inline conu/apidefs/container.py]: A question - is it possible that a container has multiple IP addresses assigned? If so, I think this should return list, but I'm not sure whether more IPs per container are even possible.

Comment (bkabrda) [inline conu/apidefs/container.py]: I also think that we should have `get_ip4` and `get_ip6`; `get_ip` should then return the same as `get_ip4`.

File: conu/apidefs/container.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    def stop(self):
+        """
+        stop this container
+
+        :return: None
+        """
+        raise NotImplementedError("stop method is not implemented")
+
+    def rm(self, force=False):
```
Comment (bkabrda) [inline conu/apidefs/container.py]: I think we should also allow removing any mounted volumes here, e.g. the same as `docker rm --volumes`.

File: conu/apidefs/container.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        """
+        copy a file or a directory from container to host system
+
+        :param src: str, path to a file or a directory within container
+        :param dest: str, path to a file or a directory on host system
+        :return: None
+        """
+        raise NotImplementedError("copy_from method is not implemented")
+
+    def read_file(self, file_path):
```
Comment (bkabrda) [inline conu/apidefs/container.py]: Would it be better to have a `File` object (that we'd also implement in conu) that could:

* be used as a context manager (e.g. `with file.read():`)
* be read in text or binary mode
* would contain all other info, like creation timestamps, permissions, etc?

Maybe I'm overengineering this, but I think it'd be nice :)

File: conu/apidefs/container.py (hunk 1 above)
Comment (TomasTomecek) [reply conu/apidefs/container.py]: It is possible. I was thinking about this actually -- not sure if it's possible for a container to have a primary IP address.

I'll update this to return list, shall we rename then to `get_IPs`?

Comment (TomasTomecek) [reply conu/apidefs/container.py]: I agree this is a good design: I fear it won't be that easy to implement it.

File: conu/apidefs/container.py (hunk 2 above)
Comment (TomasTomecek) [reply conu/apidefs/container.py]: This is actually a more generic question: what if the functionality you propose is specific to docker? shall we add `**kwargs` here? (or even to all methods?)

File: conu/apidefs/container.py (hunk 3 above)
Comment (TomasTomecek) [reply conu/apidefs/container.py]: so this be `def get_file(self, file_path):`? I like it.

File: conu/apidefs/container.py (hunk 1 above)
Comment (bkabrda) [reply conu/apidefs/container.py]: Sounds good to me.

Comment (bkabrda) [reply conu/apidefs/container.py]: So maybe we just want to have a `get_IP4` or similar, that would return list of IPv4 addresses and we don't need to care about the rest ATM. Let's just design the API in a way that will allow us to do this in future if we need to.

File: conu/apidefs/container.py (hunk 2 above)
Comment (bkabrda) [reply conu/apidefs/container.py]: Good point. I think adding `**kwargs` makes sense.

File: conu/apidefs/container.py (hunk 3 above)
Comment (bkabrda) [reply conu/apidefs/container.py]: Yes, exactly. And then users could do e.g.
```
with container.get_file('/usr/bin/foo').open('rb') as f:
    do_something()
```

Comment (TomasTomecek) [reply conu/apidefs/container.py]: Love it!

**Comment by dhodovsk** [comment]
LGTM! 
Maybe later we could add kill - for sending signals and commit/checkout for - creating snapshots.

**Comment by TomasTomecek** [comment]
I will add kill, that's a good suggestion.

**Comment by bkabrda** [comment]
Looks good, let's merge this.

**Pull request merged**